pkg/machines/components/nicBody.jsx

```diff
@@ -0,0 +1,184 @@
+            { name: 'ethernet', desc: 'Generic ethernet connection', disabled: true },
+            { name: 'direct', desc: 'Direct attachment' },
+        ];
+    else
+        availableNetworkTypes = [
+            { name: 'network', desc: 'Virtual network' },
+            { name: 'user', desc: 'Userspace SLIRP stack' },
+        ];
+
+    // Bring to the first position in dropdown list the initial selection which reflects the current nic type
+    availableNetworkTypes.sort(function(x, y) { return x.name == defaultNetworkType ? -1 : y.name == defaultNetworkType ? 1 : 0 });
+
+    if (["network", "direct"].includes(dialogValues.networkType)) {
+        if (dialogValues.networkType === "network")
+            availableSources = networks.map(network => network.name);
+        else if (dialogValues.networkType === "direct")
+            availableSources = networkDevices;
+
+        if (availableSources.length > 0) {
+            networkSourcesContent = availableSources
+                    .map(networkSource => {
+                        return (
+                            <Select.SelectEntry data={networkSource} key={networkSource}>
+                                {networkSource}
+                            </Select.SelectEntry>
+                        );
+                    });
+        } else {
+            if (dialogValues.networkType === "network")
+                defaultNetworkSource = _("No Virtual Networks");
+            else if (dialogValues.networkType === "direct")
+                defaultNetworkSource = _("No Network Devices");
+
+            networkSourcesContent = (
+                <Select.SelectEntry data='empty-list' key='empty-list'>
+                    {defaultNetworkSource}
+                </Select.SelectEntry>
+            );
+            networkSourceEnabled = false;
+        }
+    }
+
+    return (
+        <>
+            <label className='control-label' htmlFor={`${idPrefix}-select-type`}>
+                {_("Interface Type")}
+            </label>
+            <Select.Select id={`${idPrefix}-select-type`}
+                           onChange={value => onValueChanged('networkType', value)}
+                           initial={defaultNetworkType}
+                           extraClass='form-control ct-form-split'>
+                {availableNetworkTypes
+                        .map(networkType => {
+                            return (
+                                <Select.SelectEntry data={networkType.name} key={networkType.name} disabled={networkType.disabled || false}>
+                                    {networkType.desc}
+                                </Select.SelectEntry>
+                            );
+                        })}
+            </Select.Select>
+            {["network", "direct"].includes(dialogValues.networkType) && (
+                <>
+                    <label className='control-label' htmlFor={`${idPrefix}-select-source`}>
+                        {_("Source")}
```

Comment:
How come source is to the right rather than below `Interface name`?
I don't have a strong opinion either way, just wanted to verify if there was any reason for it.

Reply:
> `Interface name` ?
Do you mean `Interface type` ? The reason is that Source is heavily dependent on Interface type (child element if you want).
Also this reuses the component from Editing the Network Interface, where Source is also next to Network Type.
So alternatively we could show hierarchy by putting Source underneath the Interface name and shifting it a bit right (same as we did with hierarchy in Creation of Virtual Networks: https://github.com/cockpit-project/cockpit/pull/12266#issuecomment-510111787)